tests/unittests/sources/test_smartos.py

```diff
@@ -783,7 +781,13 @@ class TestJoyentMetadataClient(FilesystemMockingTestCase):
     def setUp(self):
         super(TestJoyentMetadataClient, self).setUp()
 
-        self.serial = mock.MagicMock(spec=serial.Serial)
+        try:
+            import serial
+        except ImportError:
+            kwargs = {}
+        else:
+            kwargs = {"spec": serial.Serial}
+        self.serial = mock.MagicMock(**kwargs)
```

Comment:
I'd like to avoid ImportError handling in unittests and ensure our unittest runner by default. We should probably make sure our py3 test runner installs all optional dependencies to ensure unittests exercise those dependencies as best as possible without custom mock handling.

 I was wondering if we want to drop this specific unittest change and instead add:
```diff
diff --git a/tox.ini b/tox.ini
index 7299b8513..7d1c1863e 100644
--- a/tox.ini
+++ b/tox.ini
@@ -20,6 +20,7 @@ passenv =
     HYPOTHESIS_PROFILE
 deps =
     -r{toxinidir}/requirements.txt
+    -r{toxinidir}/requirements-all.txt
     -r{toxinidir}/test-requirements.txt
```

Reply:
Done, thanks for the suggestion.